pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -159,60 +159,63 @@ set -e
 
 KEK_NAME=%s
 KEY_PATH=%s
-VAULT_DEFAULT_KV_VERS=v1
 CURL_PAYLOAD=$(mktemp)
 ARGS=(--silent --show-error --request GET --header "X-Vault-Token: ${VAULT_TOKEN}")
+PYTHON_DATA_PARSE="['data']"
 
 # If a vault namespace is set
 if [ -n "$VAULT_NAMESPACE" ]; then
-	ARGS+=(--header "X-Vault-Namespace: ${VAULT_NAMESPACE}")
+  ARGS+=(--header "X-Vault-Namespace: ${VAULT_NAMESPACE}")
 fi
 
 # If SSL is configured but self-signed CA is used
 if [ -n "$VAULT_SKIP_VERIFY" ] && [[ "$VAULT_SKIP_VERIFY" == "true" ]]; then
-	ARGS+=(--insecure)
+  ARGS+=(--insecure)
 fi
 
 # TLS args
 if [ -n "$VAULT_CACERT" ]; then
-	ARGS+=(--capath $(dirname "${VAULT_CACERT}"))
+  ARGS+=(--capath $(dirname "${VAULT_CACERT}"))
 fi
 if [ -n "$VAULT_CLIENT_CERT" ]; then
-	ARGS+=(--cert "${VAULT_CLIENT_CERT}")
+  ARGS+=(--cert "${VAULT_CLIENT_CERT}")
 fi
 if [ -n "$VAULT_CLIENT_KEY" ]; then
-	ARGS+=(--key "${VAULT_CLIENT_KEY}")
+  ARGS+=(--key "${VAULT_CLIENT_KEY}")
 fi
 
-# For a request to any host/port, connect to VAULT_TLS_SERVER_NAME:request's original port instead
+# For a request to any host/port, connect to VAULT_TLS_SERVER_NAME:requests original port instead
 # Used for SNI validation and correct certificate matching
 if [ -n "$VAULT_TLS_SERVER_NAME" ]; then
-	ARGS+=(--connect-to ::"${VAULT_TLS_SERVER_NAME}":)
+  ARGS+=(--connect-to ::"${VAULT_TLS_SERVER_NAME}":)
 fi
 
 # Check KV engine version
-if [ -z "$VAULT_BACKEND" ]; then
-	VAULT_BACKEND=$VAULT_DEFAULT_KV_VERS
+if [ -n "$VAULT_BACKEND" ]; then
```

Comment:
Do you need this `if` statement? Seems like checking for "v2" on the next line is sufficient by itself.